evm/src/executor/inspector/cheatcodes/expect.rs

```diff
@@ -220,6 +230,54 @@ fn expect_safe_memory(state: &mut Cheatcodes, start: u64, end: u64, depth: u64)
     Ok(Bytes::new())
 }
 
```

Comment:
can we please add a few docs here what this checks exactly checks and when this is supposed to return a revert

Reply:
done in https://github.com/foundry-rs/foundry/pull/4912/commits/ce54d4b4c820e385e539d4371e6a2367498a8790